### PR TITLE
Enable HCS for AWS

### PR DESCRIPTION
**Pull request by fhlavac**
For now, removed links to docs until HCS is GA

https://issues.redhat.com/browse/RHCLOUD-20983

Manual mode:
[screen-capture (1).webm](https://user-images.githubusercontent.com/50696716/224315093-27fb1413-dfde-4ae5-a8ca-a4adca3bcf33.webm)

Account auth.:
[screen-capture (2).webm](https://user-images.githubusercontent.com/50696716/224315277-03337321-b9c5-4a6b-a735-092fbe76988f.webm)

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@fhlavac can add the .cache folder to gitignore?

**Comment by nlcwong** [comment]
My fault! I didn't catch it in my mocks until I saw the implementation, but on Step 4, where the user is selecting the HCS bundle, I realized that my two mocks (unselected HCS & selected HCS) had two different explanation strings and you had used the first mock which had the old string version.
Can you replace:
- "Track Red Hat spend with regardless of point of purchase" with "Track Red Hat committed spend"
- "Unlock cloud images in Microsoft Azure and bring your own subscription instead of paying hourly" with "Track spend through <cloud provider type> and apply them to your Red Hat committed spend." where <cloud provider type> would be AWS, Microsoft Azure, or Google Cloud Platform, depending on what they chose in Step 1?

**Comment by nlcwong** [comment]
Also, i just wanted to make this clear that we should ONLY see the HCS bundle if the customer is participating in the HCS program (which I believe there is a flag to check for it), otherwise, it'll default to the regular cost management flow that we have today, correct?

**Comment by fhlavac** [comment]
> Also, i just wanted to make this clear that we should ONLY see the HCS bundle if the customer is participating in the HCS program (which I believe there is a flag to check for it), otherwise, it'll default to the regular cost management flow that we have today, correct?

Yeah, that's how it works. I'll update the strings. Thank you 🙂

File: src/components/FormComponents/SourceWizardSummary.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ import {
 } from '../../components/addSourceWizard/schemaBuilder';
 import { useFlag } from '@unleash/proxy-client-react';
 
-const alertMapper = (appName, sourceType, intl) => {
-  if (appName === COST_MANAGEMENT_APP_NAME && sourceType !== 'google') {
+const alertMapper = (appName, sourceType, intl, hcsEnrolled) => {
```
Comment (Hyperkid123) [inline src/components/FormComponents/SourceWizardSummary.js:31]: Can this be a regular React component? The argument list is starting to grow and its being used as a ReactNode

File: src/components/addSourceWizard/descriptions/HybridCommittedSpendDescription.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import Point from './Point';
+import { ACCOUNT_AUTHORIZATION } from '../../constants';
+
+const HybridCommittedSpendDescription = ({ id }) => {
+  const intl = useIntl();
+  const { getState } = useFormApi();
+
+  const values = getState().values;
+
+  const isEnabled =
```
Comment (Hyperkid123) [inline src/components/addSourceWizard/descriptions/HybridCommittedSpendDescription.js:16]: Can you wrap this variable init in a `useMemo` hook? We don't want to run the same condition on each render. I don't expect the ID to change very often.

File: src/components/addSourceWizard/hardcodedComponents/aws/arn.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import PropTypes, { bool } from 'prop-types';
```
Comment (Hyperkid123) [inline src/components/addSourceWizard/hardcodedComponents/aws/arn.js:2]: Probably unintentional named import?

**Comment by nlcwong** [comment]
@fhlavac Use the names you have in your source type definition.

**Comment by fhlavac** [comment]
image
updated strings & resolved comments

cc @Hyperkid123  @nlcwong

**Review by Hyperkid123 — APPROVED** [review]
LGTM

**Pull request merged**